test/Algebra/Graph/Test/Label.hs

```diff
@@ -0,0 +1,147 @@
+{-# LANGUAGE OverloadedLists #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Label
+-- License    : MIT (see the file LICENSE)
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.Label".
+-----------------------------------------------------------------------------
+module Algebra.Graph.Test.Label (
+  -- * Testsuite
+  testLabel
+  ) where
+
+import Algebra.Graph.Test hiding (NonNegative)
+import Algebra.Graph.Test.Generic (size10)
+import Algebra.Graph.Label
+import Data.Monoid
+
+type UnaryFn a = a -> a
+type BinaryFn a = a -> a -> a
+type Plus a = BinaryFn a
+type Mult a = BinaryFn a
+type Star a = UnaryFn a
+type Identity a = a
+type Zero a = a
+type One a = a
+type Annihilator a = a
+
+associative :: Eq a => BinaryFn a -> a -> a -> a -> Property
+associative f a b c = (a `f` b) `f` c == a `f` (b `f` c) // "Associative property"
+
+commutative :: Eq a => BinaryFn a -> a -> a -> Property
+commutative f a b = a `f` b == b `f` a // "Commutative property"
+
+idempotent :: Eq a => BinaryFn a -> a -> Property
+idempotent f a = a `f` a == a // "Idempotence property"
+
+annihilator :: Eq a => BinaryFn a -> Annihilator a -> a -> Property
+annihilator f z a = conjoin
+  [ a `f` z == z // "Annihilator property right"
+  , z `f` a == z // "Annihilator property left" ]
+
+closure :: Eq a => Plus a -> Mult a -> One a -> Star a -> a -> Property
+closure p m o s a = conjoin
+  [ s a == o `p` (a `m` s a) // "Star property right"
+  , s a == o `p` (s a `m` a) // "Star property left" ]
+
+leftDistributive :: Eq a => Plus a -> Mult a -> a -> a -> a -> Property
+leftDistributive p m a b c = a `m` (b `p` c) == (a `m` b) `p` (a `m` c) // "Left distributive"
```

Comment:
Just a thought: you could probably use operators for better readability, as follows:

```haskell
leftDistributive (*) (+) a b c = a * (b + c) == (a * b) + (a * c) // "Left distributive"

```

Reply:
Unfortunately, this trick doesn't work with `0` and `1` :)

Reply:
@snowleopard I wonder why it won't work for 0 and 1? Do you mean something like `leftDistributive (<*>) (<+>) 0 1 1` won't work as expected?

Reply:
I mean, you cannot implement `identity` as 
```haskell
identity (+) 0 x = (0 + x == x) && (x + 0 == x)
```
because when we use `0` on the left hand side we are comparing to zero instead of binding the input parameter to a new variable named `0` :)

Reply:
@snowleopard I am inclined to make this change. But there are also some properties with a single operator. I'd like to change those as well to maintain a sense of similarity. I don't want to use specific operators like `(*)` and `(+)`. Should I use `(..)`? If not, please suggest another operator.

Reply:
I think `*` works in most cases (e.g. `x * z == z`) apart perhaps from idempotence, where one could use something neutral like `<>` since the operation in question is typically a semigroup.